documentation/modules/oauth/proc-oauth-authentication-broker-config.adoc

```diff
@@ -134,7 +134,7 @@ Depending on how you apply OAuth 2.0 authentication, and the type of authorizati
 <5> (Only applicable when using `introspectionEndpointUri`) Depending on the authorization server you are using, the introspection endpoint may or may not return the _token type_ attribute, or it may contain different values. You can specify a valid token type value that the response from the introspection endpoint has to contain.
 <6> (Only applicable when using `introspectionEndpointUri`) The authorization server may be configured or implemented in such a way to not provide any identifiable information in an Introspection Endpoint response. In order to obtain the user ID, you can configure the URI of the `userinfo` endpoint as a fallback. The `userNameClaim`, `fallbackUserNameClaim`, and `fallbackUserNamePrefix` settings are applied to the response of `userinfo` endpoint.
 <7> Set this to `false`to disable the OAUTHBEARER mechanism on the listener. At least one of PLAIN or OAUTHBEARER has to be enabled. Default is `true`.
-<8> Set this to `true` to enable the PLAIN mechanism on the listener, which is supported by all clients on all platforms. The Kafka client has to enable the PLAIN mechanism and set the `username` and the `password`. This mechanism can be used to authenticate either by using the OAuth access token, or by using the OAuth client id and secret (client credentials). If the client sets `password` to start with the string `$accessToken:`, the password is interpreted as the access token on the server, and `username` as the account username, otherwise the user is interpreted as the client id, and password as the client secret. Default is `false`.
+<8> Set this to `true` to enable the PLAIN mechanism on the listener, which is supported by all clients on all platforms. The Kafka client has to enable the PLAIN mechanism and set the `username` and the `password`. This mechanism can be used to authenticate either by using the OAuth access token, or by using the OAuth `clientId` and `secret` (client credentials). The behavior is additionally controlled by whether `tokenEndpointUri` is specified or not. If `tokenEndpointUri` is specified, then if the client sets `password` to start with the string `$accessToken:`, the password is interpreted as the access token on the server, and `username` as the account username, otherwise the `username` is interpreted as the `clientId`, and `password` as the client `secret`, which the broker uses to obtain the access token in client's name. However, if `tokenEndpointUri` is not specified, the `password` is always interpreted as an access token, and the `username` is always interpreted as the account username which has to match the principal id extracted from the token (a so called 'no-client-credentials' mode, because client always has to obtain the access token by itself, and can't use `clientId` and `secret`). Default is `false`.
 <9> This has to be set to support the client credentials authentication when `enablePlain` is set to true, as described in previous point.
```

Comment:
@mstruk - I think number 9 also needs to be changed slightly.

Reply:
@laidan6000 Ah, I see what you mean, it's now optional so 'has to be set' is not ok. 
```suggestion
<9> Additional configuration for PLAIN mechanism to allow clients to authenticate by passing `clientId` and `secret` as `username` and `password` as described in the previous point. If not specified the clients can authenticate over PLAIN only by passing an access token as `password` parameter.
```

Reply:
Looks good.